common/nonce/errors.go

```diff
@@ -0,0 +1,6 @@
+package nonce
```

Comment:
It's nice to bring nonce items from account to common package. It's going to be very useful.
However, I don't think it needs a entire package for it. I believe the items in common package should stay on it. We will avoid 
`common.xyz.xyz` . As I mentioned before I think w should get GETH as reference.

Reply:
@jeffprestes and @arnaubennassar, explaining better the idea:

The main idea here is to actually have the types isolated in highly specialized packages so that common users know exactly where to find what.

Since everything is in common there is no specialization of the package. It's like a big disorganized toolbox where we find hammer, screwdrivers, and that kitchen knife we ​​once lost.

I understand that dividing just by dividing is foolishness and an excess of preciosity, but in this case, we are giving the end user, who does not have all the knowledge we have about the platform, a hierarchical toolbox, where he can easily find what he is looking for.

This approach **is the same as used in the standard go library**, which does not have a common package. Everything there is inside packages. And there you don't have `common.NonceFromBytes`. You have `nonce.FromBytes`, such as `ascii85.Decode` or `hex.Decode`, not `encoding.AscII85Decode` or `encoding.HexDecode` (note that nonce in this case is not a variable but the package).

This approach makes it much easier for a newcomer to our framework to use our common package. And, in the model I proposed, the occurrence of `nonce.Nonce` exists only in the type declaration. So, the main idea here is have, for a hypothetical structure Potato  the types, related errors, marshallers, etc, in a package called potato with mostly of the methods attached to the Potato struct, but when is not possible (like create a Potato from another struct, for example, PotatoFromTomato - instead having common.PotatoFromTomato we can have potato.FromTomato wich is much more declarative). 

Let me know your thoughts. But again, I ask you to look at this implementation not as hermez developers with a skewed view, but as an outsider who will use the common package.

Reply:
I got your point, and there is some cases that falls into what you're mentioning such as:

```go
package common

type PoolL2TxState string

const (
	// PoolL2TxStatePending represents a valid L2Tx that hasn't started the
	// forging process
	PoolL2TxStatePending PoolL2TxState = "pend"
	// PoolL2TxStateForging represents a valid L2Tx that has started the
	// forging process
	PoolL2TxStateForging PoolL2TxState = "fing"
	// PoolL2TxStateForged represents a L2Tx that has already been forged
	PoolL2TxStateForged PoolL2TxState = "fged"
	// PoolL2TxStateInvalid represents a L2Tx that has been invalidated
	PoolL2TxStateInvalid PoolL2TxState = "invl"
)
```

Certainly this would look better:

```go
package pooll2tx

type State string

const (
	// StatePending represents a valid L2Tx that hasn't started the
	// forging process
	StatePending PoolL2TxState = "pend"
	// ...
)
```

However moving all the types into sub-packages feels too much for me. Maybe we can find a balanced solution where we a few sub-packages that make sense among them? 

Also note that if we follow your approach we have to be very careful about cyclic dependencies

Reply:
I agree with @arnaubennassar . We need to have a balance. Maybe for items related to important business items a specific package makes sense but for accessory items I believe a toolbox is not a problem.